### PR TITLE
Fix a bug related to build TF Lite on RPI Zero.

### PR DESCRIPTION
**Pull request by marcin-sielski**
Why:

* Enable to build TF Lite on RPI Zero.

This change addresses the need by:

* Changing compiler from arm-linux-gnueabi- to arm-linux-gnueabihf-.

### CODE REVIEW

**Comment by marcin-sielski** [comment]
@googlebot I signed it!

**Comment by hajuho** [comment]
Building TF Lite on RPI Zero was successful with arm-linux-gnueabi- prefix to me.
Did you install crossbuild-essential-armel toolchain as guided at https://www.tensorflow.org/lite/guide/build_rpi#cross-compile_for_raspberry_pi ?

If arm-linux-gnueabi- is not successful even with crossbuild-essential-armel package installed, could you give me the details on your host environment?

**Comment by marcin-sielski** [comment]
Hi @hajuho,

I was building the package on the target device (Rasbian buster) and not cross-compiling on the host device.

Best Regards

Marcin Sielski

**Comment by hajuho** [comment]
Hi Terry,
Could you take a look at this?
Look like arm-linux-gnueabi- is not working for native compiling on RPI zero. Is there any problem using arm-linux-gnueabihf- ?

**Comment by marcin-sielski** [comment]
Hi,

arm-linux-gnueabihf- cross compiler available on Ubuntu will not work because it is tight to armv7. The official cross-compiler for RPI Zero is available at https://github.com/raspberrypi/tools/tree/master/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf.

Hope it helps.

Best Regards

**Comment by redsigma** [comment]
Raspberry pi zero w user. Running debian stretch 9.11 and arm-linux-gnueabi- was not preinstalled but arm-linux-gnueabihf- was. So switching to it worked.

**Comment by terryheo** [comment]
The code was added by https://github.com/tensorflow/tensorflow/issues/36445#issuecomment-585738062 and it's needed for cross compiling.
Could you update your PR to select TARGET_TOOLCHAIN_PREFIX  by checking $(HOST_ARCH) to figure out if it's cross build or native build?

**Comment by terryheo** [comment]
If you're using native build, could you try the following?

```bash
make -f tensorflow/lite/tools/make/Makefile
```

We don't need to use build script for the native build.

**Comment by marcin-sielski** [comment]
@terryheo 
Cross compiling with arm-linux-gnueabi- (armel) will work but is not efficient. The binaries that are being produced are incompatible with Rasbian binaries. They are however compatible with Debian build. The difference is in support of hard float (supported by FPU - faster) vs. soft float (emulated on CPU - slower).

**Comment by gbaned** [comment]
@terryheo  Any update on this PR? Thanks!

**Comment by terryheo** [comment]
@marcin-sielski could you confirm if this patch works for cross compiling?
Also https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/g3doc/guide/build_rpi.md should be updated.

**Comment by marcin-sielski** [comment]
@terryheo ,
This patch should work with official Raspberry Pi cross compiler. I will double confirm that. I agree that the instruction should be updated along with this change. I was not aware that the instruction is also maintained in this repository.

**Comment by terryheo** [comment]
@marcin-sielski, cool. Could you update PR with the doc change?

**Comment by marcin-sielski** [comment]
@terryheo 
I have updated this PR with instruction updates.

Best Regards

Marcin Sielski

File: tensorflow/lite/g3doc/guide/build_rpi.md [hunk 1]
```diff
@@ -5,87 +5,89 @@ Raspberry Pi. If you just want to start using TensorFlow Lite to execute your
...
-sudo apt-get update
-sudo apt-get install crossbuild-essential-armhf
-# The following is only needed for Pi Zero build.
-sudo apt-get install crossbuild-essential-armel
-```
+1. Clone official Raspberry Pi cross-compilation toolchain:
 
-If you are using Docker, you may not use `sudo`.
+    ```bash
+    git clone --depth 1 https://github.com/raspberrypi/tools.git rpi_tools
```
Comment (terryheo) [inline tensorflow/lite/g3doc/guide/build_rpi.md:26]: can we skip "--depth 1" ?

File: tensorflow/lite/g3doc/guide/build_rpi.md [hunk 2]
```diff
@@ -5,87 +5,89 @@ Raspberry Pi. If you just want to start using TensorFlow Lite to execute your
...
-./tensorflow/lite/tools/make/download_dependencies.sh
-```
+    **Note:** If you're using the TensorFlow Docker image, the repo is already provided in `/tensorflow_src/`.
 
-Note that you only need to do this once.
+3. Run following script at the root of the TensorFlow repository to download all the
+build dependencies:
+
+    ```bash
+    cd tensor_src && ./tensorflow/lite/tools/make/download_dependencies.sh
```
Comment (terryheo) [inline tensorflow/lite/g3doc/guide/build_rpi.md:42]: please correct directory name.

File: tensorflow/lite/g3doc/guide/build_rpi.md [hunk 3]
```diff
@@ -5,87 +5,89 @@ Raspberry Pi. If you just want to start using TensorFlow Lite to execute your
...
 
-This should compile a static library in:
-`tensorflow/lite/tools/make/gen/rpi_armv6/lib/libtensorflow-lite.a`.
+    **Note:** This should compile a static library in:
+    `tensorflow/lite/tools/make/gen/rpi_armv6/lib/libtensorflow-lite.a`.
 
 ## Compile natively on Raspberry Pi
 
-This has been tested on Raspberry Pi 3b, Raspbian GNU/Linux 9.1 (stretch), gcc version 6.3.0 20170516 (Raspbian 6.3.0-18+rpi1).
+Instruction has been tested on Raspberry Pi 3b, Raspbian GNU/Linux 9.1 (stretch), gcc version 6.3.0 20170516 (Raspbian 6.3.0-18+rpi1):
```
Comment (terryheo) [inline tensorflow/lite/g3doc/guide/build_rpi.md:67]: Is the gcc version the same? "gcc version 6.3.0 20170516"

**Review by terryheo — COMMENTED** [review]
Thanks for your contribution.

File: tensorflow/lite/g3doc/guide/build_rpi.md (hunk 1 above)
Comment (marcin-sielski) [reply tensorflow/lite/g3doc/guide/build_rpi.md]: fixed

File: tensorflow/lite/g3doc/guide/build_rpi.md (hunk 2 above)
Comment (marcin-sielski) [reply tensorflow/lite/g3doc/guide/build_rpi.md]: fixed

File: tensorflow/lite/g3doc/guide/build_rpi.md (hunk 3 above)
Comment (marcin-sielski) [reply tensorflow/lite/g3doc/guide/build_rpi.md]: fixed

**Comment by marcin-sielski** [comment]
@terryheo 

I think I have fixed all the issues

**Review by terryheo — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**